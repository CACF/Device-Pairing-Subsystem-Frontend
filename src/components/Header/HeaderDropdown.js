import React, {Component} from 'react';
import {
  DropdownItem,
  DropdownMenu,
  DropdownToggle,
  Dropdown
} from 'reactstrap';

class HeaderDropdown extends Component {

  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      dropdownOpen: false
    };
  }

  toggle() {
    this.setState({
      dropdownOpen: !this.state.dropdownOpen
    });
  }

  dropAccnt() {
    return (
      <Dropdown nav isOpen={this.state.dropdownOpen} toggle={this.toggle} className='dd-itereg'>
        <DropdownToggle nav>
            Hi, <span className="mr-3 h6 d-inline-block mt-2">{((this.props.userDetails || {}).preferred_username || '')}</span>
          <span className="fa fa-caret-down"></span>
        </DropdownToggle>
        <DropdownMenu right>
          <DropdownItem header tag="div" className="hide-up-md text-center">Hi, <strong>{((this.props.userDetails || {}).preferred_username || '')}</strong></DropdownItem>
          {/*<DropdownItem header tag="div" className="text-center"><strong>Account</strong></DropdownItem>*/}
          {/*<DropdownItem><i className="fa fa-user"></i> Profile</DropdownItem>*/}
          <DropdownItem onClick={this.props.kc.logout}><i className="fa fa-lock"></i> Logout</DropdownItem>
        </DropdownMenu>
      </Dropdown>
    );
  }

  render() {
    return (
      this.dropAccnt()
    );
  }
}

export default HeaderDropdown;
