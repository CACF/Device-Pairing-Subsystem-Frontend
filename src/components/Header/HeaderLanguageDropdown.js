import React, {Component} from 'react';
import {
    DropdownItem,
    DropdownMenu,
    DropdownToggle,
    Dropdown
} from 'reactstrap';

class HeaderLanguageDropdown extends Component {

    constructor(props) {
        super(props);

        this.toggle = this.toggle.bind(this);
        this.state = {
            dropdownOpen: false
        };
    }

    toggle() {
        this.setState({
            dropdownOpen: !this.state.dropdownOpen
        });
    }

    dropAccnt() {
        let lang = 'English';
        if(this.props.i18n.language === 'en') {
            lang = 'English';
        } else if(this.props.i18n.language === 'es') {
            lang = 'Español';
        } else if(this.props.i18n.language === 'id') {
            lang = 'Bahasa Indonesia';
        }
        return (
            <Dropdown nav isOpen={this.state.dropdownOpen} toggle={this.toggle}>
                <DropdownToggle caret>
                    {lang}
                </DropdownToggle>
                <DropdownMenu>
                    <DropdownItem onClick={() => this.props.switchLanguage('en')}>
                        <i className="flag-icon flag-icon-us"></i> English</DropdownItem>
                    <DropdownItem onClick={() => this.props.switchLanguage('es')}>
                        <i className="flag-icon flag-icon-es"></i> Español</DropdownItem>
                    <DropdownItem onClick={() => this.props.switchLanguage('id')}>
                        <i className="flag-icon flag-icon-id"></i> Bahasa Indonesia</DropdownItem>
                </DropdownMenu>
            </Dropdown>
        );
    }

    render() {
        return (
            this.dropAccnt()
        );
    }
}

export default HeaderLanguageDropdown;
