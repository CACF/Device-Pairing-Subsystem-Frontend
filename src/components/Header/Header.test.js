import React from 'react';
import {shallow, mount} from 'enzyme';
import Header from "./Header";

const userDetails = {
  preferred_username: "User"
}

describe('Header component',() => {

  /* Test if render successfully */
  test('Render Header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  /* Test if header renders again */
  test('Renders Header again',() => {
    const wrapper = shallow(<Header />);
    expect(wrapper.exists()).toBe(true);
  });

  /* Test if header consists of class */
  test('If header has class',() => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('.app-header').length).toEqual(1)
    expect(wrapper.find('.navbar').length).toEqual(1)
  });

  /* Test NavbarBrand render */
  test('NavbarBrand should have contains navbar-brand-minimized and navbar-brand-full',() => {
    const wrapper = shallow(<Header />);
    expect(wrapper.contains(<h5 className="navbar-brand-minimized">DPS</h5>)).toBe(true);
    expect(wrapper.contains(<h5 className="navbar-brand-full">Device Pairing Subsystem</h5>)).toBe(true);
  });

  /* Test if header consists of toggler */
  test('If header has navbar-toggler-icon',() => {
    const wrapper = shallow(<Header />);
    expect(wrapper.contains(<span className="navbar-toggler-icon"></span>)).toBe(true);
  });

  /* Test mobile sidebar button click */
  test('Mobile sidebar button click works', () => {
    const wrapper = shallow(<Header />);
    // mobile sidebar toggler button click
    wrapper.find('NavbarToggler').at(0).simulate('click',{
      preventDefault: () => {
      }
    });
    expect(document.body.classList.contains('sidebar-mobile-show'));

  });

  /* Test toggle sidebar button click */
  test('should toggle sidebar button', () => {
    const wrapper = shallow(<Header />);
    wrapper.find('NavbarToggler').at(1).simulate('click', {
        preventDefault: () => {
      }
    });
    expect(document.body.classList.contains('sidebar-hidden'))
  });


})