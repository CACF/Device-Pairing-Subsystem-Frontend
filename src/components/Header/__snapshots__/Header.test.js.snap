// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Render Header correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavbarToggler
          className="d-lg-none"
          onClick={[Function]}
          tag="button"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </NavbarToggler>,
        <NavbarBrand
          href="#"
          tag="a"
        >
          <h5
            className="navbar-brand-minimized"
          >
            DPS
          </h5>
          <h5
            className="navbar-brand-full"
          >
            Device Pairing Subsystem
          </h5>
        </NavbarBrand>,
        <NavbarToggler
          className="d-none"
          onClick={[Function]}
          tag="button"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </NavbarToggler>,
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <HeaderDropdown />
        </Nav>,
      ],
      "className": "app-header navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span
            className="navbar-toggler-icon"
          />,
          "className": "d-lg-none",
          "onClick": [Function],
          "tag": "button",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "navbar-toggler-icon",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <h5
              className="navbar-brand-minimized"
            >
              DPS
            </h5>,
            <h5
              className="navbar-brand-full"
            >
              Device Pairing Subsystem
            </h5>,
          ],
          "href": "#",
          "tag": "a",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "DPS",
              "className": "navbar-brand-minimized",
            },
            "ref": null,
            "rendered": "DPS",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Device Pairing Subsystem",
              "className": "navbar-brand-full",
            },
            "ref": null,
            "rendered": "Device Pairing Subsystem",
            "type": "h5",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span
            className="navbar-toggler-icon"
          />,
          "className": "d-none",
          "onClick": [Function],
          "tag": "button",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "navbar-toggler-icon",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <HeaderDropdown />,
          "className": "ml-auto",
          "navbar": true,
          "tag": "ul",
          "vertical": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavbarToggler
            className="d-lg-none"
            onClick={[Function]}
            tag="button"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </NavbarToggler>,
          <NavbarBrand
            href="#"
            tag="a"
          >
            <h5
              className="navbar-brand-minimized"
            >
              DPS
            </h5>
            <h5
              className="navbar-brand-full"
            >
              Device Pairing Subsystem
            </h5>
          </NavbarBrand>,
          <NavbarToggler
            className="d-none"
            onClick={[Function]}
            tag="button"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </NavbarToggler>,
          <Nav
            className="ml-auto"
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <HeaderDropdown />
          </Nav>,
        ],
        "className": "app-header navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="navbar-toggler-icon"
            />,
            "className": "d-lg-none",
            "onClick": [Function],
            "tag": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navbar-toggler-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h5
                className="navbar-brand-minimized"
              >
                DPS
              </h5>,
              <h5
                className="navbar-brand-full"
              >
                Device Pairing Subsystem
              </h5>,
            ],
            "href": "#",
            "tag": "a",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "DPS",
                "className": "navbar-brand-minimized",
              },
              "ref": null,
              "rendered": "DPS",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Device Pairing Subsystem",
                "className": "navbar-brand-full",
              },
              "ref": null,
              "rendered": "Device Pairing Subsystem",
              "type": "h5",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="navbar-toggler-icon"
            />,
            "className": "d-none",
            "onClick": [Function],
            "tag": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navbar-toggler-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <HeaderDropdown />,
            "className": "ml-auto",
            "navbar": true,
            "tag": "ul",
            "vertical": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
